lib/persistentqueue/persistentqueue.go

```diff
@@ -202,7 +204,17 @@ func tryOpeningQueue(path, name string, chunkFileSize, maxBlockSize, maxPendingB
 		fs.MustWriteAtomic(filepath, nil, false)
 	}
 	if mi.Name != q.name {
-		return nil, fmt.Errorf("unexpected queue name; got %q; want %q", mi.Name, q.name)
+		// For compatibility, further check if their name without query params and fragment would match.
+		// https://github.com/VictoriaMetrics/VictoriaMetrics/issues/8477
+		if !IsQueueNameEqual(mi.Name, q.name) {
```

Comment:
I think we agree on not letting `showURL` flag to affect the `name` in `metadata.json`. But for the hash collision I think it's still needed to add a simple check which could be not very strict. Please advise.

![image](https://github.com/user-attachments/assets/0c042edc-9054-455b-bf4e-99734a73b4ae)

Reply:
If for some reason hashing of remoteWrite.url will lead to hash collision, there is already `argIdx` as a part of queue path. And it's safe to do not perform any check on queue.Name. 

 Moreover, fastQueue package must know nothing about any remoteWrite url and it's naming format.